plugins/__init__.py

```diff
@@ -147,3 +150,116 @@ def get_resource_attribute_ref(
     :param attribute_path: The path, in the resource state dict, to the desired value.
     """
     return resource_attribute_reference(resource, attribute_path).to_dict()
+
+
+@plugin
+def serialize_config(config_block: "terraform::config::Block") -> "dict":  # type: ignore
+    """
+    Serialize a config block into a dictionnary.
+    """
+    # Build the base dict, containing all the attribute of this block
```

Comment:
it might provide a slight performance boots to go over all children, so the compiler knows which values are required

```suggestion
   for child in config_block.children:
        # access all required attributes to let the compiler know we need them 
        child.name
        child._config

    # Build the base dict, containing all the attribute of this block
```

Reply:
(this is why we do it in the exporter on the yang module)

Reply:
Interesting, thanks for the suggestion!